### PR TITLE
feat: (IAC-878) support external cds postgres

### PR DESCRIPTION
**Pull request by dhoucgitter**
### Changes
Along with the Crunchy 5 changes made in 2022.10, a blanket support statement was made that CDS Postgres should be internal only. Other teams now want to use CDS externally so allow DAC to configure and deployments to use external default and cds-postgres PostgreSQL instances for 2022.12 cadence and later.

- Change message boundary to pre 2022.12 cadences for cds-postgres instance must be internal only
- Generate user.env, secrets generator and dataserver-transformer for an external cds-postgres instance
- Modify dataserver-transformer template so that it can generate sas-platform-postgres-dataserver-transformer and sas-cds-postgres-dataserver-transformer outputs with expected dataserver names
- Include cds-postgres overlay for an external cds-postgres instance
- Use SharedServices as default cds-postgres database name when no setting is provided

### Tests
See internal ticket for additional details.

| Scenario | Task |Provider |K8s| V4_DEPLOYMENT_OPERATOR_ENABLED | Deployment Method | Viya TLS| Cadence | Notes |
|---------|--------------|----------|----------|----------|---------------|----|--------|----------------|
| 2    | External cds-postgres pre 2022.12 |Azure | v1.23.14 | false  | Ansible | full-stack | 2022.11    | As expected DAC displays the following message and exits: "CDS postgres instance must be internal server. To use CDS postgres, please reconfigure your infrastructure to set all your postgres servers internal"  |
| 3    |External cds-postgres 2022.12 |Azure | v1.23.14 | false  | Ansible | full-stack|2022.12    | As expected deployment is allowed, both default-dataserver-transformer, cds-postgres-dataserver-transformer are generated with expected content.  |
|4| fast:2020 VA-VDMML default & cds-postgres external | AWS|1.23.14 |false |Ansible|full-stack|fast:2020|Deployment stabilized with all pods Running, configured the same custom db name in ansible-vars for each PG server|
|5| risk-cirrus, TLS disabled |AWS|1.23.14|false|Ansible|disabled|stable:2023.01|All sas-risk-cirrus pods stabilized and running. Verified sas-risk-data pod successful connection to external cds-postgres instance and SCHEMA creation.|
|7|risk-cirrus, TLS enabled |AWS|1.23.14|false|Ansible|front-door|stable:2023.01|Deployment stabilized, all sas-risk-cirrus-pods running. sas-risk-data pod log shows successful connection to ssl_enabled external cds-postgres instance and SCHEMA creation.|
|8| risk-cirrus, TLS enabled |Azure|1.24.6|false|Ansible| front-door|stable:2023.01| sas-risk-data pod log shows successful connection to external cds-postgres instance and SCHEMA creation. Deployment stabilized with all risk-cirrus pods Running. |
|9|risk-cirrus, Deployment Operator, TLS enabled |Azure|1.24.6|true|Ansible| front-door|stable:2023.01| DO enabled. Deployment stabilized, all sas-risk-cirrus-pods Running. sas-risk-data pod log showing successful connection to ssl_enabled external cds-postgres instance and SCHEMA creation.|

### CODE REVIEW

**Review by riragh — APPROVED** [review]

**Review by jarpat — APPROVED** [review]

File: roles/vdm/tasks/postgres/postgres-instance.yaml [hunk 1]
```diff
@@ -145,6 +145,15 @@
 
 - name: postgres instance - external post 2022.10
```
Comment (thpang) [inline roles/vdm/tasks/postgres/postgres-instance.yaml:146]: Name seems a bit off now given we also have 2022.12 in the mix.

**Review by thpang — CHANGES_REQUESTED** [review]
Except for task name all looks good.

File: roles/vdm/tasks/postgres/postgres-instance.yaml (hunk 1 above)
Comment (dhoucgitter) [reply roles/vdm/tasks/postgres/postgres-instance.yaml:146]: @thpang, would you suggest adding 2022.12 into the mix or just removing cad ver from comment entirely?

File: roles/vdm/tasks/postgres/postgres-instance.yaml [hunk 2]
```diff
@@ -143,7 +143,7 @@
     - uninstall
     - update
 
-- name: postgres instance - external post 2022.10
+- name: postgres instance - external post 2022.10 - external cds 2022.12
```
Comment (dhoucgitter) [inline roles/vdm/tasks/postgres/postgres-instance.yaml:146]: @thpang, how's this updated task name?

Comment (thpang) [reply roles/vdm/tasks/postgres/postgres-instance.yaml:146]: Maybe something like: `external cds post 2022.12`. Just that small change and it'll be good to go

**Review by thpang — CHANGES_REQUESTED** [review]

**Review by thpang — APPROVED** [review]

**Pull request merged**